Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -122,3 +164,30 @@ incrementOrSetTotaRides driverId driverStats = do
       driverStats.totalRidesAssigned
   Esq.runNoTransaction $ DSQ.incrementTotalRidesAssigned (cast driverId) incrementTotaRidesBy
   pure $ driverStats {DS.totalRidesAssigned = Just $ fromMaybe 0 driverStats.totalRidesAssigned + incrementTotaRidesBy}
+
+getDriverStats :: (EsqDBFlow m r, EsqDBReplicaFlow m r) => Maybe DS.DriverStats -> Id DP.Person -> Maybe Money -> m DS.DriverStats
+getDriverStats Nothing driverId _ = createDriverStat driverId
+getDriverStats (Just driverStats) driverId rideFare = do
+  updatedTotalRideCount <- getTotalRideCount
+  cancelledCount <-
+    case driverStats.ridesCancelled of
+      Nothing -> Esq.runInReplica $ BCRQ.findAllCancelledByDriverId driverId
+      Just cancelledCount -> pure $ cancelledCount + 1
```

Comment:
why +1?

Reply:
As we discussed, This +1 is for incrementing the cancelled ride count, and this is implemented by Piyush, I just refactored this part.